nuxeo-retention/src/main/java/org/nuxeo/retention/service/RetentionManagerImpl.java

```diff
@@ -279,68 +312,92 @@ public void evalRules(Map<String, Set<String>> docsToCheckAndEvents) {
         Framework.getService(WorkManager.class).schedule(work, WorkManager.Scheduling.ENQUEUE);
     }
 
-    protected ELActionContext initActionContext(DocumentModel doc, CoreSession session) {
-        ELActionContext ctx = new ELActionContext(new ExpressionContext(), new ExpressionFactoryImpl());
-        ctx.setCurrentPrincipal(session.getPrincipal());
-        doc.detach(true);
-        ctx.setCurrentDocument(doc);
-        return ctx;
-    }
-
-    protected boolean evaluateConditionExpression(ELActionContext ctx, String expression) {
-        Calendar now = Calendar.getInstance();
+    protected boolean evaluateConditionExpression(Record record, String eventInput, CoreSession session) {
+        String expression = record.getRule(session).getStartingPointExpression();
         if (StringUtils.isEmpty(expression)) {
-            return false;
+            return true;
         }
+        ELActionContext ctx = new ELActionContext(new ExpressionContext(), new ExpressionFactoryImpl());
+        record.getDocument().detach(true);
+        ctx.setCurrentDocument(record.getDocument());
+        Calendar now = Calendar.getInstance();
         ctx.putLocalVariable("currentDate", now);
+        ctx.putLocalVariable(EVENT_BASED_RULE_INPUT, eventInput);
+        // Only restricted user can fill expression and eventInput
         return ctx.checkCondition(expression);
     }
 
     @Override
-    public void evalExpressionEventBasedRules(Record record, Set<String> events, CoreSession session) {
-        if (record == null) {
-            return; // nothing to do
-        }
+    public boolean applyEventBasedRules(Record record, String event, String eventInput, CoreSession session) {
```

Comment:
Maybe put in a var `rule.getDocument().getPathAsString()` and `record.getDocument().getPathAsString()`, used several times in the method?